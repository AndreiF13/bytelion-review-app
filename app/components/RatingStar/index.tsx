import React from 'react';
// Components
import { View, Text, Image, InteractionManager } from 'react-native';
import { MaterialCommunityIcons } from '@expo/vector-icons';

export interface IRatingStarProps {
  rating: number;
  starSize: number;
}

export default function RatingStar(props: IRatingStarProps ) {
  // Creating a simple array to simplify the process of drawing the stars
  // This array will be used to map and to check if rating inside the range
  const arrStars = [
    { from: 1 },
    { from: 2 },
    { from: 3 },
    { from: 4 },
    { from: 5 },
  ];

  return (
    <View style={{ flexDirection: "row" }}>
      {arrStars.map((item, index) => (
        <MaterialCommunityIcons key={index} name="star" size={props.starSize} color={props.rating >= item.from ? "#FFC107" : "#BDBDBD"} />
      ))}      
    </View>
  )
}