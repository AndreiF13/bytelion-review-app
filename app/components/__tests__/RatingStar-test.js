import * as React from 'react';
import renderer from 'react-test-renderer';
import RatingStar from '../RatingStar';

it(`RatingStar component should render correctly`, () => {
    const tree = renderer.create(<RatingStar rating={3.5} starSize={28} />).toJSON();
    expect(tree).toMatchSnapshot();
});
