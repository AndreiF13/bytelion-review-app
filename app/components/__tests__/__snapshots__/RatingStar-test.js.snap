// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingStar component should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text />
  <Text />
  <Text />
  <Text />
  <Text />
</View>
`;
